Isosurfacing_3/test/Isosurfacing_3/test_topology.cpp

```diff
@@ -0,0 +1,658 @@
+
+    Point_range points_high_res;
+    Triangle_range triangles_high_res;
+    IS::marching_cubes<CGAL::Parallel_if_available_tag>(domain_high_res, iso, points_high_res, triangles_high_res, CGAL::parameters::use_topologically_correct_marching_cubes(true));
+
+    #ifdef CGAL_TESTSUITE_ISOSURFACING_OUTPUT
+    CGAL::IO::write_polygon_soup("trilinear.off", points_high_res, triangles_high_res);
+    #endif
+
+    assert(CGAL::Polygon_mesh_processing::is_polygon_soup_a_polygon_mesh(triangles_high_res));
+    Mesh m;
+    CGAL::Polygon_mesh_processing::polygon_soup_to_polygon_mesh(points_high_res, triangles_high_res, m);
+
+    std::cout << "components: " << connected_components(m) << std::endl;
+    std::cout << "euler: " << euler_characteristic(m) << std::endl;
+    std::cout << "boundaries: " << boundary_components(m) << std::endl;
+  }
+
+  {
+    Point_range points_mc;
+    Triangle_range triangles_mc;
+    IS::marching_cubes<CGAL::Parallel_if_available_tag>(domain, iso, points_mc, triangles_mc, CGAL::parameters::use_topologically_correct_marching_cubes(false));
+
+    #ifdef CGAL_TESTSUITE_ISOSURFACING_OUTPUT
+    CGAL::IO::write_polygon_soup("mc.off", points_mc, triangles_mc);
+    #endif
+  }
+
+  Point_range points;
+  Triangle_range triangles;
+  IS::marching_cubes<CGAL::Sequential_tag>(domain, iso, points, triangles, CGAL::parameters::use_topologically_correct_marching_cubes(true));
+
+  #ifdef CGAL_TESTSUITE_ISOSURFACING_OUTPUT
+  CGAL::IO::write_polygon_soup("tmc.off", points, triangles);
+  #endif
+}
+
+template<typename KERNEL>
+void assert_tmc(const std::array<typename KERNEL::FT, 8>& case_values, typename KERNEL::FT iso, std::size_t components, std::size_t euler, std::size_t boundaries)
+{
+  using K = KERNEL;
+  using Grid = IS::Cartesian_grid_3<K>;
+  using Values = IS::Interpolated_discrete_values_3<Grid>;
+  using Domain = IS::Marching_cubes_domain_3<Grid, Values>;
+  using Point = typename K::Point_3;
+  using Mesh = CGAL::Surface_mesh<Point>;
+  using Point_range = std::vector<Point>;
+  using Triangle_range = std::vector<std::vector<std::size_t> >;
+
+  {
+    Grid grid { Point{-1., -1., -1.}, Point{1., 1., 1.}, std::array<std::size_t, 3>{2, 2, 2} };
+    Values values { grid };
+    Domain domain { grid, values };
+
+    to_grid(grid, values, case_values);
+
+    Point_range points;
+    Triangle_range triangles;
+    IS::marching_cubes<CGAL::Sequential_tag>(domain, iso, points, triangles, CGAL::parameters::use_topologically_correct_marching_cubes(true));
+
+    assert(points.size() && triangles.size());
+    assert(!has_duplicate_points(points, triangles));
+    assert(!has_duplicate_polygons(points, triangles));
+    assert(!has_isolated_vertices(points, triangles));
+
+    assert(CGAL::Polygon_mesh_processing::is_polygon_soup_a_polygon_mesh(triangles));
+    Mesh m;
+    CGAL::Polygon_mesh_processing::polygon_soup_to_polygon_mesh(points, triangles, m);
+
+    assert(connected_components(m) == components);
```

Comment:
`warning: comparison of integers of different signs: 'int' and 'std::size_t' (aka 'unsigned long') [-Wsign-compare]`

Reply:
for the 3 assertions